pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,700 @@
+
+        # Get biases
+        b1 = trg.b1(self.z_s)
+        b2 = trg.b2(self.z_s)
+        bs = trg.bs(self.z_s)
+        bk2 = trg.bk2(self.z_s)
+        b3nl = trg.b3nl(self.z_s)
+        if (np.any(b2 != 0) or np.any(bs != 0) or
+                np.any(bk2 != 0) or np.any(b3nl != 0)):
+            warnings.warn(
+                "EulerianPTCalculators assume linear galaxy bias "
+                "when computing galaxy-IA cross-correlations.",
+                category=CCLWarning)
+        c1 = tri.c1(self.z_s)
+        c2 = tri.c2(self.z_s)
+        cd = tri.cdelta(self.z_s)
+
+        pgi = b1[:, None] * (c1[:, None] * Pd1d1 +
+                             (self._g4*cd)[:, None] * (a00e + c00e)[None, :] +
+                             (self._g4*c2)[:, None] * (a0e2 + b0e2)[None, :])
+        return pgi*self.exp_cutoff
+
+    def _get_pgm(self, trg):
+        """ Get the number counts - matter cross-spectrum at the internal
+        set of wavenumbers and scale factors.
+
+        Args:
+            trg (:class:`~pyccl.nl_pt.tracers.PTTracer`): number
+                counts tracer.
+
+        Returns:
+            array_like: 2D array of shape `(N_a, N_k)`, where `N_k` \
+                is the size of this object's `k_s` attribute, and \
+                `N_a` is the size of the object's `a_s` attribute.
+        """
+        self._check_pt_init()
+        # Get Pk templates
+        Pd1d1 = self.pk_b1
+        Pd1d2 = self._g4[:, None] * self.dd_bias[2][None, :]
+        Pd1s2 = self._g4[:, None] * self.dd_bias[4][None, :]
+        Pd1p3 = self._g4[:, None] * self.dd_bias[8][None, :]
+        Pd1k2 = self.pk_bk*(self.k_s**2)[None, :]
+
+        # Get biases
+        b1 = trg.b1(self.z_s)
+        b2 = trg.b2(self.z_s)
+        bs = trg.bs(self.z_s)
+        bk2 = trg.bk2(self.z_s)
+        b3nl = trg.b3nl(self.z_s)
+
+        pgm = (b1[:, None] * Pd1d1 +
+               0.5 * b2[:, None] * Pd1d2 +
+               0.5 * bs[:, None] * Pd1s2 +
+               0.5 * b3nl[:, None] * Pd1p3 +
+               0.5 * bk2[:, None] * Pd1k2)
+
+        return pgm*self.exp_cutoff
+
+    def _get_pii(self, tr1, tr2, return_bb=False):
+        """ Get the intrinsic alignment auto-spectrum at the internal
+        set of wavenumbers and scale factors.
+
+        Args:
+            tr1 (:class:`~pyccl.nl_pt.tracers.PTTracer`): first tracer
+                to correlate.
+            tr2 (:class:`~pyccl.nl_pt.tracers.PTTracer`): first tracer
+                to correlate.
+
+        Returns:
+            array_like: 2D array of shape `(N_a, N_k)`, where `N_k` \
+                is the size of this object's `k_s` attribute, and \
+                `N_a` is the size of the object's `a_s` attribute.
+        """
+        self._check_pt_init()
+        # Get Pk templates
+        Pd1d1 = self.pk_b1
+        a00e, c00e, a0e0e, a0b0b = self.ia_ta
+        ae2e2, ab2b2 = self.ia_tt
+        a0e2, b0e2, d0ee2, d0bb2 = self.ia_mix
```

Comment:
These variable names are 
![Untitled](https://user-images.githubusercontent.com/22447385/230157113-4b7e10e6-69e0-4070-83b0-2230f1ff1eeb.jpeg)
ae2200e2aae02ea2ae00ea2

`".join(np.random.choice(["a", "e", "0", "2"], 23))`

Reply:
I don't disagree, but this is very similar to what FASTPT does internally, so I'd rather keep them to make any future comparisons with the raw code of FASTPT easier.

Reply:
Yes, I didn't suggest changing them, I just commented on how terrible they are xD